src/jit/morph.cpp

```diff
@@ -10026,22 +10013,6 @@ GenTree* Compiler::fgMorphBlockOperand(GenTree* tree, var_types asgType, unsigne
         {
             lclNode = effectiveVal->AsLclVarCommon();
         }
-#ifdef FEATURE_SIMD
-        if (varTypeIsSIMD(asgType))
-        {
-            if ((indirTree != nullptr) && (lclNode == nullptr) && (indirTree->Addr()->OperGet() == GT_ADDR) &&
-                (indirTree->Addr()->gtGetOp1()->OperIsSimdOrHWintrinsic()))
-            {
-                assert(!isDest);
-                needsIndirection = false;
-                effectiveVal     = indirTree->Addr()->gtGetOp1();
-            }
-            if (effectiveVal->OperIsSimdOrHWintrinsic())
-            {
-                needsIndirection = false;
-            }
-        }
-#endif // FEATURE_SIMD
```

Comment:
This code is also not directly related to this fix, but is a path that will never be hit because we would be on the other path of the outer if in this case.